private_gpt/components/node_store/node_store_component.py

```diff
@@ -16,19 +17,38 @@ class NodeStoreComponent:
     doc_store: BaseDocumentStore
 
     @inject
-    def __init__(self) -> None:
-        try:
-            self.index_store = SimpleIndexStore.from_persist_dir(
-                persist_dir=str(local_data_path)
-            )
-        except FileNotFoundError:
-            logger.debug("Local index store not found, creating a new one")
-            self.index_store = SimpleIndexStore()
-
-        try:
-            self.doc_store = SimpleDocumentStore.from_persist_dir(
-                persist_dir=str(local_data_path)
-            )
-        except FileNotFoundError:
-            logger.debug("Local document store not found, creating a new one")
-            self.doc_store = SimpleDocumentStore()
+    def __init__(self, settings: Settings) -> None:
+        match settings.docstore.database:
+            case "local":
+                try:
+                    self.index_store = SimpleIndexStore.from_persist_dir(
+                        persist_dir=str(local_data_path)
+                    )
+                except FileNotFoundError:
+                    logger.debug("Local index store not found, creating a new one")
+                    self.index_store = SimpleIndexStore()
+
+                try:
+                    self.doc_store = SimpleDocumentStore.from_persist_dir(
+                        persist_dir=str(local_data_path)
+                    )
+                except FileNotFoundError:
+                    logger.debug("Local document store not found, creating a new one")
+                    self.doc_store = SimpleDocumentStore()
+
+            case "postgres":
+                from llama_index.core.storage.index_store.postgres_index_store import PostgresIndexStore
```

Comment:
This needs to be wrapped in a try - catch, because otherwise PrivateGPT would not work if the dependency is not installed. Check vector store component for example. Something like this:

```
                try:
                    from llama_index.vector_stores.postgres import (  # type: ignore
                        PGVectorStore,
                    )
                except ImportError as e:
                    raise ImportError(
                        "Postgres dependencies not found, install with `poetry install --extras vector-stores-postgres`"
                    ) from e
```

Reply:
postgres is a subset of the pgvector requirements.  To keep things simple, I'll create the dependency to vector-stores-postgres extra rather than introduce another.

Reply:
I don't fully follow. I'd separate dependencies as much as possible, to allow for example using Postgres for index and doc store but not as vector store.

Reply:
You'll note that I have created a single dependency ```storage-postgres``` for index/docstore postgres supporta.   This is due to the fact that both of these are used together, but they are separated in llama-index.  Also I have created dependencies on two PG libraries.  This is because the llama-index postgres component don't do it.  They should.  I need to raise a pull against llama-index to fix that. To prevent this project failing I've included them.